components/policy-mgt/org.wso2.carbon.policy.mgt.core/src/main/java/org/wso2/carbon/policy/mgt/core/mgt/impl/PolicyManagerImpl.java

```diff
@@ -726,4 +726,17 @@ public boolean setPolicyApplied(DeviceIdentifier deviceIdentifier) throws Policy
             throw new PolicyManagementException(msg, e);
         }
     }
+
+    @Override
+    public int getPolicyCount() throws PolicyManagementException {
+        int policyCount = 0;
+        try {
+            policyCount = policyDAO.getPolicyCount();
```

Comment:
Also, in some other code you've written, which appears right at the top of this pull request, "this" is used to refer to global variables maintained within the respective classes. However, here you seem to have not done that. Even though there's no hard and fast rule on this, might probably be a good idea to stick to one approach just for the purpose of making your code appear consistent across the whole code-base.